cirq-core/cirq/ops/raw_types.py

```diff
@@ -853,17 +851,16 @@ def _validate_qid_shape(val: Any, qubits: Sequence['cirq.Qid']) -> None:
         ValueError: The operation had qids that don't match it's qid shape.
     """
     qid_shape = protocols.qid_shape(val)
-    if len(qubits) != len(qid_shape):
+    qubits_shape = protocols.qid_shape(qubits)
```

Comment:
I have a slight preference to revert to the previous implementation here because the `qid_shape` protocol seems to be slow, and we know here that it's the same as `tuple(q.dimension for q in qubits)` but we don't actually need to construct the tuple.

Reply:
reverted